src/ui-test/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,201 @@
+
+const TEST_FOLDER = '../../../testFolder';
+const WORKSPACE_FOLDER = path.join(__dirname, TEST_FOLDER);
+
+const START_DEBUG_LABEL = 'Start Java debugger on Camel K integration';
+const REMOVE_INTEGRATION_LABEL = 'Remove Apache Camel K Integration';
+
+
+describe('Tooling for Apache Camel K extension', function () {
+
+	before(async function () {
+		this.timeout(200000);
+		await prepareTempWorkspaceForTests(WORKSPACE_FOLDER);
+		await VSBrowser.instance.driver.sleep(3000);
+	});
+	
+	describe('Java Debug', function () {
+		
+		const INTEGRATION_LABEL = 'java-debug-test';
+		const INTEGRATION_FILE = 'JavaDebugTest';
+
+		before(async function (){
+			this.timeout(20000);
+			await createIntegration(INTEGRATION_FILE);
+			await startIntegrationOnCurrentFile();
+			await VSBrowser.instance.driver.sleep(5000);
+		})
+
+		it('Check Java Debug available', async function () {
+			this.timeout(20000);
+
+			const item = await findIntegrationOnSideBar(INTEGRATION_LABEL);
+			const menu = await item.openContextMenu();
+
+			assert.isTrue(await menu.hasItem(START_DEBUG_LABEL));
+		});
+
+		it.skip('Check Java Debug stops at breakpoint', async function() {
+			//Currently failing because "No Java Debugger configured" error window
+			//It won't work even if I install the extension using ExTester or manually on the VSCode Instance
+			this.timeout(30000);
+			await addBreakpointToFile();
+			await startDebuggerOn(INTEGRATION_LABEL);
+			await VSBrowser.instance.driver.sleep(3000);
+			assert.isTrue(isThereSomethingInVariablesInDebugger());
+
+			await stopAndExitDebug();
+		})
+
+		after(async function() {
+			this.timeout(20000);
+			await removeIntegration(INTEGRATION_LABEL);
+			await prepareEmptyTestFolder(WORKSPACE_FOLDER);
+		});
+
+	});
+
+	describe('No Java Debug on Invalid Files', function() {
+
+		const INTEGRATION_LABEL = 'java-debug-test-invalid';
+		const INTEGRATION_FILE = 'JavaDebugTestInvalid';
+
+		before(async function (){
+			this.timeout(20000);
+			await createIntegration(INTEGRATION_FILE);
+			await modifyCurrentFileToBeInvalid();
+			await startIntegrationOnCurrentFile();
+			await VSBrowser.instance.driver.sleep(3000);
+		});
+
+		it('Test Java Debugger Not Available On Invalid File', async function() {
+			this.timeout(20000)
+			const item = await findIntegrationOnSideBar(INTEGRATION_LABEL);
+			const menu = await item.openContextMenu();
+
+			assert.isFalse(await menu.hasItem(START_DEBUG_LABEL));
+		});
+
+		after(async function() {
+			this.timeout(20000);
+			await removeIntegration(INTEGRATION_LABEL);
+			await prepareEmptyTestFolder(WORKSPACE_FOLDER);
+		});
+	})
+
+});
+async function startIntegrationOnCurrentFile() {
+	const workbench = new Workbench();
+	await workbench.executeCommand('Start Apache Camel K Integration');
+	console.log('Start command');
+	const startMode = await InputBox.create();
+	await startMode.selectQuickPick('Basic');
+
+	await VSBrowser.instance.driver.sleep(1000);
```

Comment:
sleep should be avoided as much as possible. it is introducing slowness and flakiness to test suite. it will be better to use a waitUntil with a specific condition